promql/engine.go

```diff
@@ -1496,8 +1503,17 @@ func (ev *evaluator) evalVectorSelector(ctx context.Context, vs *parser.VectorSe
 				if ss.Floats == nil {
 					ss.Floats = reuseOrGetFPointSlices(prevSS, numSteps)
 				}
-				ss.Floats = append(ss.Floats, FPoint{F: f, T: ts})
+				val := f
+				if recordOrigT {
+					// This is an info metric, where we want to track the original sample timestamp.
```

Comment:
nit: could note that all values in an info metric are expected to be 1, thus we can re-use the space.

Reply:
Done.